api-server/src/events/events.service.ts

```diff
@@ -89,18 +89,24 @@ export class EventsService {
       );
 
       // Sort events by the first start_time and apply pagination
+      // Note, we have to sort by first_start_time in our common table expression to apply pagination correctly, but we
+      // also have to sort again over our page in the final join with events, because the order after joining is not
+      // guaranteed to stay the same.
+      // We've also added a secondary order by on created_at to ensure that events with with not start_time like
+      // online resources at least sort consistently
       const paginatedRows: EventModel[] = await this.sequelize.query(
         `
               with compressed_event as (SELECT events.*, min(dv.start_time) as first_start_time
                                         FROM events
                                                  LEFT OUTER JOIN datetime_venue dv on events.id = dv.event_id
                   ${whereClause}
               GROUP BY (events.id)
-              ORDER BY first_start_time DESC
+              ORDER BY first_start_time DESC, "createdAt" DESC
               OFFSET ${(requestedPage - 1) * pageSize} LIMIT ${pageSize} )
               SELECT events.*
               FROM compressed_event
                        JOIN events ON events.id = compressed_event.id
+                       ORDER BY compressed_event.first_start_time DESC, "createdAt" DESC
```

Comment:
this addtion of  `ORDER BY compressed_event.first_start_time DESC` here on the outer fetch is the main fix here, adding in `"createdAt" DESC` is some icing on the cake to further lock in the order

Reply:
Why are these sorts `DESC`? I'm sure this is right, given we haven't noticed any issues, but I would have assumed we'd want to sort by ascending dates?

Reply:
I get it now -- this isn't the current-events endpoint, it's _all the events_. Makes more sense, for the moment, to get more recent stuff first vs events from 6+ years ago.